pkg/virt-manifest/mapper.go

```diff
@@ -42,52 +43,40 @@ type savedDisk struct {
 	disk v1.Disk
 }
 
-func ExtractPvc(dom *v1.DomainSpec) (*v1.DomainSpec, []savedDisk) {
-	specCopy := &v1.DomainSpec{}
-	model.Copy(specCopy, dom)
-
+func ExtractPvc(dom *v1.DomainSpec) []savedDisk {
 	pvcDisks := []savedDisk{}
-	allDisks := []v1.Disk{}
 
-	for idx, disk := range specCopy.Devices.Disks {
+	for idx, disk := range dom.Devices.Disks {
 		if disk.Type == Type_PersistentVolumeClaim {
 			// Save the disk so we can fix it later
 			diskCopy := v1.Disk{}
 			model.Copy(&diskCopy, disk)
 			pvcDisks = append(pvcDisks, savedDisk{disk: diskCopy, idx: idx})
-
-			// Alter the disk record so that libvirt will accept it
-			disk.Type = Type_Network
-			disk.Source.Protocol = "iscsi"
 		}
-		allDisks = append(allDisks, disk)
 	}
-	// Replace the Domain's disks with modified records
-	specCopy.Devices.Disks = allDisks
 
-	return specCopy, pvcDisks
+	return pvcDisks
 }
 
 // This is a simplified version of the domain creation portion of SyncVM. This is intended primarily
 // for mapping the VM spec without starting a domain.
-func MapVM(con virtwrap.Connection, vm *v1.VM) (*v1.VM, error) {
+func MapVM(con virtwrap.Connection, vm *v1.VM, k8sClient kubernetes.Interface) (*v1.VM, error) {
```

Comment:
Could not directly reply to https://github.com/kubevirt/kubevirt/pull/331#discussion_r129267237, therefore here:

Could you provide more examples like this? As you know that I am very interested in this defaulting, such examples are very valuable for me. Right now, we don't support anything like this vhost-user, and I would want to look at this, when we want to support it.  Finally it looks like that is really a special interface, so I would expect, that one hotplugs or adds exactly such a device then (so it would be on the VM object). I would at least first question, if we want to abstract that away.

Again, I am very interested in such special cases, and would want to investigate these and value your input. Right now, we don't need the kubernetes connection there, and PVCs don't necessarily exist at this stage, therefore trying to resolve them is pointless. Not sure if something like that exists, but if we start supporting network resources which we don't attach as block device, it might be necessary to revisit that. Right now, I didn't see any special configuration on disk devices (regarding to bus, pci addresses, ...; with other words what the VM sees), no matter if ceph, gluster or iscsi, where we need to resolve the source.